site/src/pages/components/switch.explainer.mdx

```diff
@@ -0,0 +1,574 @@
+| `value`     | `"on"`/`"off"`(?) | `"off"`       | Determines the current value of the component.                                                                        | Changed upon user interaction on click, touch, space key (when focused), enter key (when focused, optional) |
+| `disabled`  | `true`/`false`    | `false`       | When `true`, disables the control, preventing user interaction and displaying the control with a disabled appearance. | No interaction. Controlled programmatically.                                                                |
+| `focused`   | `true`/`false`    | `false`       |                                                                                                                       |                                                                                                             |
+
+### Accessibility
+
+The control can be in a checked or unchecked state. [^6]
+
+The `<switch>` element can be assigned the default role of switch as described in the [Core Accessibility API Mappings](https://w3c.github.io/core-aam/#role-map-switch). [^7]
+
+Content for thumb and either track side can be defined by the author.
+
+#### Keyboard Navigation and Focus
+
+{/* - _Arrow vs tabbing key behavior_ */}
+{/* - _Modifier key effects (e.g. shift, ctrl, meta)_ */}
+{/* - _Does this component use focus delegation?_ */}
+
+| Key              | Description                            |
+| ---------------- | -------------------------------------- |
+| Space            | Should toggle the switch when focused. |
+| Enter (optional) | Should toggle the switch when focused. |
+
+#### Form Input
+
+{/* _Does this component integrate with form submission, form validation, etc.? Can integration be accomplish with standard APIs or are special work-arounds required?_ */}
+
+The switch is also a form element, acting as a mix between a button and a checkbox - a stateful button.
+
+##### Type button
+
+A JavaScript event listener is used to trigger the immediate action assigned to the switch. This can include calling a REST API or toggling a class on an element (e.g. dark mode).
+
+##### Type submit
+
+<div class="question">Feedback on this feature would be welcome.</div>
+
+For a more traditional, non-JavaScript approach, it's necessary to receive the state of the switch on the server. In order to facilitate the directness of the action, a switch can be of type "submit". Like a submit button, toggling the switch will immediately submit the associated form to the server. This enables creating a HTML/CSS only settings page while staying true to user expectations of how the component should behave.
+
+#### Use with Assistive Technology
+
+{/* _Are there any details about shadow dom, focus delegation, or aria attributes that need special attention?_ */}
+
+The switch element can be associated with the existing `role="switch"`.
```

Comment:
rather than commenting on the type bits above, i'll just repeat here that while a switch generally performs an immediate action due to the scripting associated with it, it is not really thought of as a type of submit button, and i've already outlined in a previous comment how submit buttons / form submissions are expected to behave - which does not match the expected UX of a switch.

Reply:
Removed that section from the explainer